src/main/kotlin/no/nav/familie/ks/sak/kjerne/beregning/TilkjentYtelseService.kt

```diff
@@ -56,15 +59,57 @@ class TilkjentYtelseService(
                 tilkjentYtelse = tilkjentYtelse,
             )
 
-        val alleAndelerTilkjentYtelse =
-            andelerTilkjentYtelseBarnaMedAlleEndringer.map { it.andel } +
-                overgangsordningAndelerSomAndelTilkjentYtelse +
-                andelerForPraksisendring2024
+        val alleOrdinæreAndeler = andelerTilkjentYtelseBarnaMedAlleEndringer.map { it.andel }
+
+        val alleOrdinæreAndelerJustert = alleOrdinæreAndeler.taHensynTilPraksisendringAndeler(andelerForPraksisendring2024)
+
+        val alleAndelerTilkjentYtelse = alleOrdinæreAndelerJustert + overgangsordningAndelerSomAndelTilkjentYtelse
 
         tilkjentYtelse.andelerTilkjentYtelse.addAll(alleAndelerTilkjentYtelse)
         return tilkjentYtelse
     }
 
+    private fun List<AndelTilkjentYtelse>.taHensynTilPraksisendringAndeler(praksisendringAndeler: List<AndelTilkjentYtelse>): List<AndelTilkjentYtelse> {
+        val ordinæreAndeler = this
+
+        if (praksisendringAndeler.isEmpty()) return ordinæreAndeler
+        if (ordinæreAndeler.isEmpty()) return praksisendringAndeler
+
+        val aktørerMedPraksisendringAndeler = praksisendringAndeler.map { it.aktør }
+        val aktørerMedOrdinæreAndeler = ordinæreAndeler.map { it.aktør }
+
+        val alleAktører = (aktørerMedPraksisendringAndeler + aktørerMedOrdinæreAndeler).distinct()
+
+        val ordinæreAndelTidslinjer = ordinæreAndeler.tilSeparateTidslinjerForBarna()
+        val praksisendringTidslinjer = praksisendringAndeler.tilSeparateTidslinjerForBarna()
+
+        return alleAktører.flatMap { aktør ->
+            val ordinæreAndelTidslinjeForAktør = ordinæreAndelTidslinjer[aktør] ?: return@flatMap praksisendringAndeler.filter { it.aktør == aktør }
+            val praksisendringTidslinjeForAktør = praksisendringTidslinjer[aktør] ?: return@flatMap ordinæreAndeler.filter { it.aktør == aktør }
+
+            ordinæreAndelTidslinjeForAktør
+                .kombinerMed(praksisendringTidslinjeForAktør) { ordinæreAndel, praksisendringAndel ->
+                    praksisendringAndel ?: ordinæreAndel
```

Comment:
Burde vi også sjekke om ordinærAndel != full sats for at vi skal returnere praksisendringAndel?

Reply:
Ser nå at det ikke genereres praksisendringAndel hvis prosent=100, så da er kanskje ikke dette så viktig